apps/aecore/src/aec_chain.erl

```diff
@@ -252,6 +273,16 @@ genesis_header() ->
 find_common_ancestor(Hash1, Hash2) when is_binary(Hash1), is_binary(Hash2) ->
     aec_chain_state:find_common_ancestor(Hash1, Hash2).
 
+
+-spec hash_is_in_main_chain(binary()) -> boolean().
+hash_is_in_main_chain(Hash) when is_binary(Hash) ->
+    case get_header(Hash) of
+        error -> false;
+        {ok, Header} ->
+            Height = aec_headers:height(Header),
+            {ok, Hash} =:= aec_chain_state:get_hash_at_height(Height)
```

Comment:
Fragile to return type change (`{ok, H}` vs. `H`) though a debatable cosmetic matter.